ProjFS.Mac/PrjFSKextTests/HandleOperationTests.mm

```diff
@@ -1129,16 +1191,21 @@ - (void) testWriteWithDisappearingVirtualizationRoot {
         [&]()
         {
             ActiveProvider_Disconnect(self->dummyRepoHandle, &self->dummyClient);
+            // mark as hydrated
+            testFileVnode->attrValues.va_flags &= ~FileFlags_IsEmpty;
```

Comment:
I guess our code doesn't test that the empty flag was removed, only whether the xattr is there, so it's not required for passing the test to remove the flag, but I'd be uneasy with the half-arsed approach.

I'm going to re-check what exactly we're using this test for though, I no longer remember after a month so that's probably a good indication that we need some comments in there.

Reply:
Yep, that sounds spot on. I've changed the xattr logic and also added some comments explaining what's being tested and how.